### PR TITLE
EREGCSC-1754 Added Secret Scanning and Readme Guide

### PR DESCRIPTION
**Pull request by maira-samtek**
Resolves #1754

**Description-**
This pull request implements secret scanning

**This pull request changes...**
-  Gitleaks pre-commit hook and github action workflow is set up
- Added Readme guide for secret scanning

**Steps to manually verify this change...**

1. Make sure to setup pre-commit and gitleaks
2. Verify github action runs 
3. Try to commit any secret on branch

### CODE REVIEW

File: .github/workflows/git-secrets.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+on: [pull_request, push, workflow_dispatch]
+jobs:
+  gitleaks-scan:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run gitlakes docker	    
+      uses: docker://zricethezav/gitleaks
+      with:
+        args: detect --source /github/workspace/ --no-git --verbose
```
Comment (PhilR8) [inline .github/workflows/git-secrets.yaml:11]: Github is showing a "no newline at tend of file" warning here.  Can your editor be configured to automatically "insert final newline" at the end of new files?  Here's a discussion about how to do it in VS Code:

https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files

**Review by PhilR8 — CHANGES_REQUESTED** [review]

File: .github/workflows/git-secrets.yaml (hunk 1 above)
Comment (maira-samtek) [reply .github/workflows/git-secrets.yaml]: @PhilR8 I added the final new lines as suggested, but now the github check failed for e2e testing, it says that cypress failed

Comment (PhilR8) [reply .github/workflows/git-secrets.yaml]: The failure is unrelated to your work here -- we're taking care of it in another story.  Thank you for making these changes!

**Review by PhilR8 — APPROVED** [review]
Tested locally on MacOS and works as expected on commit.  Pushed commit up to source control and Github Action works as expected.  LGTM!

**Pull request merged**